src/crypto/crypto_common.cc

```diff
@@ -323,8 +323,8 @@ constexpr auto GetCipherVersion = GetCipherValue<SSL_CIPHER_get_version>;
 StackOfX509 CloneSSLCerts(X509Pointer&& cert,
                           const STACK_OF(X509)* const ssl_certs) {
   StackOfX509 peer_certs(sk_X509_new(nullptr));
-  if (cert)
-    sk_X509_push(peer_certs.get(), cert.release());
+  if (!peer_certs || (cert && !sk_X509_push(peer_certs.get(), cert.release())))
+    return StackOfX509();
```

Comment:
Maybe split this into two if statements for legibility (they're separate actions) but otherwise LGTM.